local-cli/eject/eject.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const copyProjectTemplateAndReplace = require('../generator/copyProjectTemplateAndReplace');
+const path = require('path');
+const fs = require('fs');
+
+function eject() {
+
+  const doesIOSExist = fs.existsSync(path.resolve('ios'));
+  const doesAndroidExist = fs.existsSync(path.resolve('android'));
+  if (doesIOSExist && doesAndroidExist) {
+    console.error('Both the iOS and Android folders already exist! Please delete `ios` and/or `android` before ejecting.');
+    process.exit();
+  }
+
+  let appConfig = null;
+  try {
+    appConfig = require(path.resolve('app.json'));
+  } catch(e) {
+    console.error('Eject requires an `app.json` config file to be located in your app root, and it must at least specify a `name` and a `displayName`.');
```

Comment:
nit: in your app root directory.

It would also be very helpful to print the absolute path (`path.resolve('app.json')`) as part of the error message to make it clear what you mean by app root directory.